inc/Settings/Settings.php

```diff
@@ -0,0 +1,151 @@
+<?php declare(strict_types = 1);
+
+namespace VIPRealTimeCollaboration\Settings;
+
+defined( 'ABSPATH' ) || exit();
+
+final class Settings {
+	private const SETTINGS_PAGE_SLUG = 'vip-real-time-collaboration-settings';
+	public const OPTION_NAME = 'vip_real_time_collaboration_settings';
+
+	public function __construct() {
+		add_action( 'admin_init', [ $this, 'register_settings' ] );
+		add_action( 'admin_menu', [ $this, 'add_options_page' ] );
+	}
+
+	public static function is_vip_rtc_enabled(): bool {
+		/** @var array<string> */
+		$options = get_option( self::OPTION_NAME, self::get_default_options() );
+
+		return isset( $options['enable-vip-rtc'] ) && (bool) $options['enable-vip-rtc'];
+	}
+
+	/**
+	 * Get the default options for the plugin.
+	 *
+	 * @return array The default options.
+	 */
+	public static function get_default_options(): array {
+		$default_options = [];
+
+		$default_options['enable-vip-rtc'] = true;
+
+		return $default_options;
+	}
+
+	/**
+	 * Sanitize settings before saving.
+	 *
+	 * @param array $input The input values from the form.
+	 * @return array The sanitized settings.
+	 * @psalm-suppress PossiblyUnusedMethod Psalm does not detect usage via add_filter.
+	 */
+	public static function sanitize_settings( ?array $input = [] ): array {
+		$sanitized = [];
+
+		// Handle checkbox - if not set in input, it means unchecked.
+		$sanitized['enable-vip-rtc'] = isset( $input['enable-vip-rtc'] ) && '1' === $input['enable-vip-rtc'];
+
+		return $sanitized;
+	}
+
+	/**
+	 * Register the settings for the settings page
+	 */
+	public static function register_settings(): void {
+		register_setting(
+			self::SETTINGS_PAGE_SLUG,
+			self::OPTION_NAME,
+			[
+				'type' => 'array',
+				'default' => self::get_default_options(),
+				'sanitize_callback' => [ __CLASS__, 'sanitize_settings' ],
+			]
+		);
+
+		// Add instructions section.
+		add_settings_section(
+			'plugin-settings',
+			'',
+			[ __CLASS__, 'display_settings_instructions' ],
+			self::SETTINGS_PAGE_SLUG
+		);
+
+		/** @psalm-suppress InvalidArgument */ // WordPress Settings API allows custom args.
+		add_settings_field(
+			'enable-vip-rtc',
+			__( 'Enable VIP Real-Time Collaboration', 'vip-real-time-collaboration' ),
+			[ __CLASS__, 'display_settings_checkbox' ],
+			self::SETTINGS_PAGE_SLUG,
+			'plugin-settings',
+			[
+				'label' => __( 'Disabling this, will disable the VIP Real-Time Collaboration plugin. This can be used in case of an emergency.', 'vip-real-time-collaboration' ),
```

Comment:
Happy to tweak this verbiage

Reply:
The radio button works a lot better than the checkbox, as I was able to change the text to be a lot less confusing. Check it out, and let me know what you think.